.github/workflows/main.yml

```diff
@@ -0,0 +1,23 @@
+name: CI
+on: [push]
+jobs:
+  build:
+    name: Test
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install Dependencies
+        uses: borales/actions-yarn@v2.0.0
+        with:
+          cmd: install # will run `yarn install` command
+      - name: Run Unit Tests
+        uses: borales/actions-yarn@v2.0.0
+        with:
+          cmd: test # will run `yarn test` command
+      - name: Run E2E Tests
+        shell: bash
+        run: |
+           yarn start:mock &
+           sleep 2m &&
+           curl http://localhost:8000 &&
```

Comment:
Instead of sleeping for 2 minutes, could we do something like: https://stackoverflow.com/questions/11904772/how-to-create-a-loop-in-bash-that-is-waiting-for-a-webserver-to-respond/50583452?

Reply:
This is the kind of thing we do in the `e2etests` script.  Is there a way to just invoke that `e2etests` script directly?

Reply:
This worked! Much cleaner, thanks @portante.

Reply:
@gurbirkalsi, you have a somewhat generous definition of "worked".  😁 

I love the simplicity of the `main.yml` file, now, but I'm concerned about merging this in while it's producing a failed result.

Reply:
OK, so it passed on its most recent run...I'm not sure what changed (I don't even see what triggered the re-run of the checks).  Are the results somehow non-deterministic?

For posterity, here's the failure which has subsequently disappeared:
```
Running E2E tests ...
yarn run v1.22.10
$ umi test ./src/e2e --verbose
FAIL src/e2e/controllers.e2e.js (13.325 s)
  controller page component
    ✓ should load controllers (5124 ms)
    ✓ should search for controller name (84 ms)
    ✓ should reset search results (66 ms)
    ✕ should sort controllers column alphabetically ascending (93 ms)
    ✓ should sort controllers column alphabetically descending (67 ms)
    ✓ should sort last modified column chronologically ascending (85 ms)
    ✓ should sort last modified column chronologically descending (71 ms)
    ✓ should sort results column numerically ascending (67 ms)
    ✓ should sort results column numerically descending (63 ms)
    ✓ should display 20 controllers in the table (53 ms)
    ✓ should display 50 controllers on preference (87 ms)
    ✓ should display 75 controllers on preference (126 ms)
    ✓ should display 100 controllers on preference (129 ms)
    ✓ should display the date picker component on click (112 ms)
    ✓ should change the selected dates after picking them from the calendar (287 ms)

  ● controller page component › should sort controllers column alphabetically ascending

    expect(received).toBe(expected) // Object.is equality

    Expected: "accusamus41"
    Received: "accusamus6"

      100 |       elem => elem.innerHTML
      101 |     );
    > 102 |     expect(testController).toBe(
          |                            ^
      103 |       mockControllers.sort((a, b) => (a.controller > b.controller ? 1 : -1))[0].controller
      104 |     );
      105 |   });

      at Object.<anonymous> (src/e2e/controllers.e2e.js:102:28)
```

Reply:
I realize that the contents (and stability) of the tests are probably out-of-scope for this PR, but I regard testing as foundational, so I'm quite uncomfortable with the notion that the tests are "passing" while issuing warnings or that they sometimes spontaneously fail.  I haven't done a lot of front-end work, so perhaps I'm just naive and that this is just the way it is...but I hope to heck that that is not the case!

So, I have two issues:  
- first, I'd love to get some reassurance the test problems that we're seeing here really do have nothing to do with the fact that they are being run under GitHub Actions (meaning that they should not gate this PR);
- second, do we have (or should someone like me) open issues against the dashboard for these warnings?  (@anishaswain, @aquibbaig, @dbutenhof, @portante)

Reply:
The previous test run failed due to duplicate data generated from the mock environment. This is an issue @aquibbaig opened here: https://github.com/distributed-system-analysis/pbench-dashboard/issues/136

Ideally, the tests expect data from our mock environment that is easily assertable. This can be improved by either running postprocessing on generated mock datasets to remove duplicates, selecting data from a "predefined" list of items, or adding unique IDs to each generated value. All approaches would allow us to avoid issues of duplication & ensure consistent tests going forward.

This should be addressed in a separate PR that includes changes to the mocked environment. For the scope of this PR, we are only moving CI functionality from TravisCI to GitHub Actions.